cvat-ui/src/containers/annotation-page/top-bar/top-bar.tsx

```diff
@@ -105,30 +125,78 @@ function mapDispatchToProps(dispatch: any): DispatchToProps {
     };
 }
 
-type Props = StateToProps & DispatchToProps;
+type Props = StateToProps & DispatchToProps & RouteComponentProps;
 class AnnotationTopBarContainer extends React.PureComponent<Props> {
+    private static beforeUnloadCallback(event: BeforeUnloadEvent): any {
+        const confirmationMessage = 'You have unsaved changes, please confirm leaving this page.';
+        // eslint-disable-next-line no-param-reassign
+        event.returnValue = confirmationMessage;
+        return confirmationMessage;
+    }
+
+    private autoSaveInterval: number | undefined;
+    private unblock: any;
+
+    componentDidMount(): void {
+        const {
+            autoSave,
+            autoSaveInterval,
+            saving,
+        } = this.props;
+
+        this.autoSaveInterval = window.setInterval((): void => {
+            if (autoSave && !saving) {
+                this.onSaveAnnotation();
+            }
+        }, autoSaveInterval);
+
+        this.checkUnsavedChanges();
+    }
+
     public componentDidUpdate(): void {
         const {
             jobInstance,
+            frameSpeed,
             frameNumber,
+            frameDelay,
             playing,
             canvasIsReady,
-            onChangeFrame,
             onSwitchPlay,
+            onChangeFrame,
         } = this.props;
 
+
         if (playing && canvasIsReady) {
             if (frameNumber < jobInstance.stopFrame) {
+                let framesSkiped = 0;
+                if (frameSpeed === FrameSpeed.Fast) {
+                    framesSkiped = 1;
+                }
+                if (frameSpeed === FrameSpeed.Fastest) {
+                    framesSkiped = 2;
+                }
+
                 setTimeout(() => {
                     const { playing: stillPlaying } = this.props;
                     if (stillPlaying) {
-                        onChangeFrame(frameNumber + 1);
+                        onChangeFrame(frameNumber + 1 + framesSkiped);
                     }
-                });
+                }, frameDelay);
             } else {
                 onSwitchPlay(false);
             }
         }
+
+        this.checkUnsavedChanges();
```

Comment:
I have several thoughts here.

First of all the method ``componentDidUpdate()`` here is called after each change of annotations because of props ``undoAction`` and ``redoAction`` are changed. It isn't very obvious, is it? If in the future we can remove them from here (forgot about this fact), and it won't work. And even now there is examples of scenario when it doesn't work:

1) Open a job
2) Move an object
3) Save the job
4) Move an object 
5) Go to task link

Result: You do not get any confirmation messages. 

It is because in both cases ``undoAction`` is a string "Changed points" and the second time component isn't going to update because it sees that there aren't updated props. 

One more concern is that ``checkUnsavedChanges()`` can be extremely slow (especially when there are a lot of annotations), but in this implementation it is going to be called after any small change. UI in this case is slow and has a lot of lags.